services/site/src/app/challenge/[challengeSlug]/page.tsx

```diff
@@ -0,0 +1,110 @@
+
+  const response = await client.query<ChallengeDetailsBySlugQuery>({
+    query: ChallengeDetailsBySlugDocument,
+    variables: { slug },
+  });
+
+  return response.data.challenge;
+}
+
+const Challenge = async ({ params: { challengeSlug } }: PageProps): Promise<React.ReactElement> => {
+  const challenge = await getChallenge(challengeSlug);
+
+  if (!challenge) {
+    notFound();
+  }
+
+  const DifficultyIcon = getDifficultyIconByChallengeDifficulty(challenge.difficulty);
+  const firstUnfinishedLevel = challenge.levels.find((level) => level.status === LevelStatus.Open || level.status === LevelStatus.InProgress);
+
+  return (
+    <>
+      <Navigation displayBreadcrumbs />
+      <main className="h-full box-border max-w-screen-3xl mx-auto">
+        <div className="mx-8 py-8 h-main max-w-screen-3xl sm:mx-12 lg:mt-12 lg:mx-24">
+          <h1 className="mb-8 h2 md:h1">
+            <span className="sr-only">Challenge: </span>
+            {challenge?.name}
+            <div className={clsx("inline-block")}>
+              <DifficultyIcon className={clsx("w-5 h-12 inline-block")} firstClassName="ml-4" />
+            </div>
+          </h1>
+          <p className="prose" dangerouslySetInnerHTML={{ __html: sanitizeHtml(challenge.introduction) }} />
+
+          <ul className="gap-4 pt-8 grid grid-cols-1 gap-y-4 sm:grid-cols-2 lg:grid-cols-3 2xl:grid-cols-4">
+            {challenge.levels.map((level, index) => {
+              const isFirstUnfinishedLevel = level.id === firstUnfinishedLevel?.id;
+
+              function getTitlePrefix(): string {
+                if (index === 0 && isFirstUnfinishedLevel) {
+                  return "Start here: ";
+                }
+
+                if (isFirstUnfinishedLevel) {
+                  return "Up next: ";
+                }
+
+                return "";
+              }
+
+              return (
+                <li className="m-0 p-0" key={level.id}>
+                  <Link
+                    href={`/challenge/${challengeSlug}/level/${String(level.order).padStart(2, "0")}`}
+                    className={clsx(
+                      "relative block border border-solid rounded-lg px-4 py-3 w-full h-18",
+                      "hover:bg-primary-dark hover:border-primary-dark",
+                      !isFirstUnfinishedLevel && "border-grey-dark",
+                      isFirstUnfinishedLevel && "border-primary bg-primary",
+                    )}
+                  >
+                    <span className="h3 text-base text-grey block">
+                      {getTitlePrefix()}Level {String(level.order).padStart(2, "0")}
+                    </span>
+                    <span className="font-normal text-grey-middle mb-0 block">{level.__typename === "QuizLevel" ? "Quiz" : "Coding"}</span>
+                    {level.status === LevelStatus.Finished && (
+                      <>
+                        <span className="sr-only">You have already completed this level.</span>
```

Comment:
I would shorten the text here as it gets very tiring with the long sentence when you already completed 6 Levels and have to go through every one.

```suggestion
                        <span className="sr-only">Completed.</span>
```